napari/_qt/_tests/test_async_slicing.py

```diff
@@ -98,18 +130,72 @@ def test_async_slice_multiscale_image_on_zoom(qtbot, make_napari_viewer, rng):
     image.corner_pixels = np.array([[0, 2, 3], [0, 6, 7]])
     image.events.reload(Event('reload', layer=image))
 
-    wait_until_vispy_image_data_equal(qtbot, vispy_image, data[0][2, 2:6, 3:7])
+    wait_until_vispy_image_data_equal(qtbot, vispy_layer, data[0][2, 2:6, 3:7])
+
+
+def test_slicing_in_progress(make_napari_viewer, qtbot, rng):
+    viewer = make_napari_viewer()
+    data = rng.random((3, 4, 5))
+    layer = LockableImage(data)
+    vispy_layer = setup_viewer_for_async_slice(viewer, layer)
+    assert viewer.dims.current_step != (2, 0, 0)
+
+    layer = vispy_layer.layer
+    assert not viewer.slicing_in_progress
+    with layer.lock:
+        viewer.dims.current_step = (2, 0, 0)
+        assert viewer.slicing_in_progress
+
+    wait_until_vispy_image_data_equal(qtbot, vispy_layer, data[2, :, :])
+    assert not viewer.slicing_in_progress
+
+
+def test_async_slice_image_loaded(make_napari_viewer, qtbot, rng):
+    viewer = make_napari_viewer()
+    data = rng.random((3, 4, 5))
+    layer = LockableImage(data)
+    vispy_layer = setup_viewer_for_async_slice(viewer, layer)
+
+    assert layer.loaded
+    assert viewer.dims.current_step != (2, 0, 0)
+
+    with layer.lock:
+        viewer.dims.current_step = (2, 0, 0)
+        assert not layer.loaded
 
+    wait_until_vispy_image_data_equal(qtbot, vispy_layer, data[2, :, :])
+    assert layer.loaded
 
-def setup_viewer_for_async_slice_image(
+
+def test_async_slice_points_loaded(make_napari_viewer, qtbot, rng):
+    viewer = make_napari_viewer()
+    shape = (10, 4)
+    np.random.seed(0)
+    data = 20 * np.random.random(shape)
+    layer = LockablePoints(data)
+    vispy_layer = setup_viewer_for_async_slice(viewer, layer)
+
+    assert layer.loaded
+    assert viewer.dims.ndisplay != 3
+    with layer.lock:
+        viewer.dims.ndisplay = 3
+        assert not layer.loaded
+
+    wait_until_vispy_image_data_equal(
+        qtbot, vispy_layer, np.flip(data[:, 1:][-3]).reshape(1, 3)
```

Comment:
This is obnoxious, but I have no idea the proper way to go from the original to the expected. Can someone help?